src/domains/transaction/hooks/use-transaction-types.ts

```diff
@@ -85,10 +85,14 @@ export const useTransactionTypes = ({ wallets = [] }: TransactionTypeProperties
 		},
 	};
 
+	const mainsailUnsupportedTypes = new Set(["htlcClaim", "htlcLock", "htlcRefund", "ipfs", "secondSignature", "magistrate"]);
```

Comment:
this shouldn't be necessary. The `wallet.transactionTypes()` should instead only list supported types. For example for Mainsail it currently lists these when you log it:
```
["delegateRegistration","usernameRegistration","usernameResignation","delegateResignation","ipfs","multiPayment","multiSignature","secondSignature","transfer","vote"]
```

This still includes IPFS and secondSignature, but that means you need to adjust this on the PSDK side to not return unsupported transaction types